lib/coinbase/wallet.rb

```diff
@@ -123,11 +123,15 @@ def wallets_api
     #   with the Wallet. If not provided, the Wallet will derive the first default address.
     # @param client [Jimson::Client] (Optional) The JSON RPC client to use for interacting with the Network
     def initialize(model, seed: nil, address_models: [])
-      validate_seed_and_address_models(seed, address_models)
+      validate_seed_and_address_models(seed, address_models) unless Coinbase.use_server_signer?
 
       @model = model
-      @master = master_node(seed)
       @addresses = []
+
+      return if Coinbase.use_server_signer?
```

Comment:
We would still want to fetch the addresses in this case potentially?

Maybe we can lazily do that when calling `addresses`, but that's one thing I'd like to think through.

Currently calling `User#wallets` has a major N+1 in terms of fetching addresses. I'd like to potentially delay that fetch until we actually need to interact with the addresses?

Reply:
@alex-stone as an interim fix, updated this logic to load addresses even while using signer.
Lets optimize the address fetching in the next release